seidr/dev.py

```diff
@@ -77,7 +77,7 @@ def beam_search(beam, update, ranking=standard_ranking, beam_width=100):
             yield code
             parents.append(code)
 
-        parents = itertools.islice(ranking(new_beam), beam_width)
+        parents = itertools.islice(ranking(parents), beam_width)
 
         if len(parents) == 0:
```

Comment:
Could you check that this `if len(parents) == 0` is needed?  It gave me an error once: one cannot check length of an iiterator